### ISSUE TITLE
AttributeError: ParaFrag instance has no attribute 'width'

### ISSUE THREAD

**Issue by belate** [rst2pdf/rst2pdf#544]
Please could someone help me with this error?

[ERROR] pdfbuilder.py:130 ParaFrag instance has no attribute 'width'
Traceback (most recent call last):
  File "/Users/belate/.virtualenvs/sphinx/lib/python2.7/site-packages/rst2pdf/pdfbuilder.py", line 127, in write
    docwriter.write(doctree, destination)
  File "/Users/belate/.virtualenvs/sphinx/lib/python2.7/site-packages/docutils/writers/**init**.py", line 80, in write
    self.translate()
  File "/Users/belate/.virtualenvs/sphinx/lib/python2.7/site-packages/rst2pdf/pdfbuilder.py", line 621, in translate
    compressed=self.compressed)
  File "/Users/belate/.virtualenvs/sphinx/lib/python2.7/site-packages/rst2pdf/createpdf.py", line 653, in createPdf
    pdfdoc.multiBuild(elements)
  File "/Users/belate/.virtualenvs/sphinx/lib/python2.7/site-packages/reportlab/platypus/doctemplate.py", line 1014, in multiBuild
    self.build(tempStory, **buildKwds)
  File "/Users/belate/.virtualenvs/sphinx/lib/python2.7/site-packages/reportlab/platypus/doctemplate.py", line 927, in build
    self.handle_flowable(flowables)
  File "/Users/belate/.virtualenvs/sphinx/lib/python2.7/site-packages/rst2pdf/createpdf.py", line 762, in handle_flowable
    if frame.add(f, canv, trySplit=self.allowSplitting):
  File "/Users/belate/.virtualenvs/sphinx/lib/python2.7/site-packages/rst2pdf/flowables.py", line 559, in add
    return Frame.add(self, flowable, canv, trySplit)
  File "/Users/belate/.virtualenvs/sphinx/lib/python2.7/site-packages/reportlab/platypus/frames.py", line 167, in _add
    w, h = flowable.wrap(aW, h)
  File "/Users/belate/.virtualenvs/sphinx/lib/python2.7/site-packages/reportlab/platypus/paragraph.py", line 1091, in wrap
    blPara = self.breakLines(self._wrapWidths)
  File "/Users/belate/.virtualenvs/sphinx/lib/python2.7/site-packages/reportlab/platypus/paragraph.py", line 1358, in breakLines
    _words[0:0] = _splitFragWord(w,maxWidth-spaceWidth-currentWidth,maxWidths,lineno)
  File "/Users/belate/.virtualenvs/sphinx/lib/python2.7/site-packages/reportlab/platypus/paragraph.py", line 510, in _splitFragWord
    for g,cw,c in _fragWordIter(w):
  File "/Users/belate/.virtualenvs/sphinx/lib/python2.7/site-packages/reportlab/platypus/paragraph.py", line 481, in _fragWordIter
    yield f, getattr(f,'width'), s
AttributeError: ParaFrag instance has no attribute 'width'
FAILED
build succeeded, 170 warnings.

Thank you very much

**Comment by pmaupin** [comment]
a) What version of reportlab are you using?  (I think it might work best with 2.6)
b) Can you give a sample that fails?

**Comment by belate** [comment]
I'm using reportlab==3.2.0

The error happen when the pages of the pdf are created:

[INFO] createpdf.py:1112 Page 1 [5]
[INFO] createpdf.py:1112 Page 2 [6]
[ERROR] pdfbuilder.py:130 ParaFrag instance has no attribute 'width'

Is the reportlab version correct?

**Comment by pmaupin** [comment]
I would suspect that 2.4 or 2.5 would work best, but don't know for sure.  A lot of stuff changed in 3.

**Comment by belate** [comment]
Ok, I will try

**Comment by belate** [comment]
Using reportlab==2.7 works!! 
Thank you very much

**Comment by pmaupin** [comment]
You're welcome!  If you have a small testcase that shows a failure, it would be nice to add it for when we start trying to support ReportLab 3 properly.

Thanks,
Pat

**Comment by sheldonhu** [comment]
i repace reportlab 3.2 with 2.7, another error occur. 
[ERROR] pdfbuilder.py:130 tostring() has been removed. Please call tobytes() instead

**Comment by sheldonhu** [comment]
i have solved the error. according to the error trace, i open reportlab/lib/util.py and modify the content.
in line 651, i modify tostr

**Comment by sheldonhu** [comment]
i have solved the error. according to the error trace, i open reportlab/lib/util.py and modify the content.
in line 651, i modify "tostring" to "tobytes".

**Comment by AndydeCleyre** [comment]
I'm experiencing this issue with `rst2pdf 0.93` and `reportlab 3.2.0`. Below is a sample document that, with a minor modification, either does or does not trigger the error.

Triggers the error:

```restructuredtext
Software Notes
--------------

+-------------+-----------------------------------------------+----------------------------------------------------------------------------------------------------------------------------+
| Software    | Version                                       | Notes                                                                                                                      |
+=============+===============================================+============================================================================================================================+
| Ubuntu      | 15.04 (Vivid Vervet) or 15.10 (Wily Werewolf) | Full-on systemd support. Investigate 16.04 LTS (Xenial Xerus) in April, being wary of Python versions and Snappy packages. |
+-------------+-----------------------------------------------+----------------------------------------------------------------------------------------------------------------------------+
| debootstrap | >= 1.0.77                                     | Generate filesystems for containers.                                                                                       |
+-------------+-----------------------------------------------+----------------------------------------------------------------------------------------------------------------------------+
```

Does *not* trigger the error:

```restructuredtext
Software Notes
--------------

+-------------+-----------------------------------------------+----------------------------------------------------------------------------------------------------------------------------+
| Softwa      | Version                                       | Notes                                                                                                                      |
+=============+===============================================+============================================================================================================================+
| Ubuntu      | 15.04 (Vivid Vervet) or 15.10 (Wily Werewolf) | Full-on systemd support. Investigate 16.04 LTS (Xenial Xerus) in April, being wary of Python versions and Snappy packages. |
+-------------+-----------------------------------------------+----------------------------------------------------------------------------------------------------------------------------+
| debootstrap | >= 1.0.77                                     | Generate filesystems for containers.                                                                                       |
+-------------+-----------------------------------------------+----------------------------------------------------------------------------------------------------------------------------+
```

**Comment by 31416r** [comment]
This is a very old thread, but as I had a very similar issue, I think it might be good to signal that this seems to be related to a bug in reportlab when handling text width (visibly when put inside tables in the rst2pdf use case) : https://bitbucket.org/rptlab/reportlab/issues/81/failed-to-access-width-attribute-in

This bug was corrected in the reportlab version 3.3.8, and thus it should be corrected as well with the reportlab release 3.4.0 (latest release at the time of this writing); at least, it worked for me.

**Comment by lornajane** [comment]
Thanks for the info!  Since it seems like we don't need to do anything, I'll close this issue.  But I've captured the examples above as a test so we can be aware of systems where this is happening.
